### DISCUSSION TITLE
[QUESTION] Using SkiaSharp with an OpenGL context

### DISCUSSION THREAD

**Discussion by FireBanana** [mono/SkiaSharp#1608] [Q&A]
Hi, I'm trying to run SkiaSharp on an OpenGL context (in a console app) but all I see is a black screen. I'm not sure what I'm doing wrong as I'm new to this. Also, I'm using a GLFW wrapper for the OpenGL calls. Any help would be appreciated.

            GLFW.glfwInit();
            GL10.glHint(GLFW.GLFW_CONTEXT_VERSION_MAJOR, 3);
            GL10.glHint(GLFW.GLFW_CONTEXT_VERSION_MINOR, 3);
            GL10.glHint(GLFW.GLFW_OPENGL_PROFILE, GLFW.GLFW_OPENGL_CORE_PROFILE);

            var window = GLFW.glfwCreateWindow(800, 600, new byte[0], IntPtr.Zero, IntPtr.Zero);
            GLFW.glfwMakeContextCurrent(window);            
            GL10.glViewport(0, 0, 800, 600); 

            SKImageInfo imageInfo = new SKImageInfo(800, 600);

            var context = GRContext.CreateGl(GRGlInterface.CreateOpenGl((name) => new IntPtr(GLFW.glfwGetProcAddress(name))));

            using (SKSurface surface = SKSurface.Create(context, false, imageInfo, 1, GRSurfaceOrigin.TopLeft))
            {
                SKCanvas canvas = surface.Canvas;

                while (GLFW.glfwWindowShouldClose(window) == 0)
                {
                    canvas.Clear(SKColors.Red);
                    GLFW.glfwSwapBuffers(window);
                    GLFW.glfwPollEvents();

                }
            }

**Comment by bmitc** [comment]
Did you ever figure this out? I think I saw (maybe it was your post elsewhere) a post about this and one suggestion was to use `canvas.Flush()`.

Which GLFW .NET wrapper were/are you using? I am trying to use Silk.NET but am having trouble.

**Reply by FireBanana** [reply]
I wasn't able to create a context from the SkiaSharp library, but I was able to create a context normally (in GLFW) and then created a texture with simple shaders, and passed a pointer to the texture that's rendered by SkiaSharp. You can look at a repo on my profile (SkiaCore) where I implemented it. The code's a bit messy at the moment but you should be able to see how it works.

The wrapper I'm using is Arqan
